File: Desenvolvimento/client/src/@fuse/types/models/empresa-endereco.ts
import { Empresa } from './empresa';
import { Endereco } from './endereco';
import { BaseModel } from '../base-model';

export class EmpresaEndereco extends BaseModel {
    constructor(
        public id: number,
        public empresaId?: number,
        public empresa?: Empresa,
        public enderecoId?: number,
        public Eendereco?: Endereco
        ) { super(); }
}

File: Desenvolvimento/server/src/ApplicationCore/Entities/EmpresaEndereco.cs
using ApplicationCore.Entities.Identity;
using ApplicationCore.Interfaces.Base;
using System;
using System.Collections.Generic;
using System.Text;

namespace ApplicationCore.Entities
{
    public class EmpresaEndereco : IBaseEntity<long>
    {
        public long Id { get; set; }

        public long? EmpresaId { get; set; }
        public virtual Empresa Empresa { get; set; }

        public long? EnderecoId { get; set; }
        public virtual Endereco Endereco { get; set; }
    }
}
